modules/command-registry.js

```diff
@@ -0,0 +1,402 @@
+        return BASE_COMMAND_DEFINITIONS;
+    },
+
+    /**
+     * Returns a registered command by name. If the command is not registered, returns `undefined`
+     *
+     * @param {string} name
+     *
+     * @returns {program.Command}
+     */
+    getCommand(name) {
+        return _getCommand(name);
+    },
+
+    /**
+     * Sets a flag to determine how base commands are registered (whether they are located in the
+     * node_modules folder, or the current project directory)
+     *
+     * Should only be called once.
+     *
+     * @param {boolean} isImportedModule
+     *
+     * @returns `this` for chaining
+     */
+    initialize(isImportedModule) {
+        _validateInitializationOrExit(isImportedModule);
+
+        _isImportedModule = isImportedModule;
+
+        return this;
+    },
+
+    /**
+     * Registers a single base command by name available from the `and-cli` with the program.
+     *
+     * Note: Prints an error if the specified command name is not found.
+     *
+     * @param {string} name Name of the base command to register
+     * @param {boolean} [overrideIfRegistered=false] If true, subsequent registrations of a command
+     * with the same name will override the last. Otherwise, a warning will be displayed and the
+     * original command will remain.
+     *
+     * @returns `this` for chaining
+     */
+    registerBaseCommand(name, overrideIfRegistered = false) {
+        const baseCommand = _validateAndGetBaseCommand(name);
+
+        if (baseCommand == null) {
+            return this;
+        }
+
+        if (!_validateCommandRegistration(baseCommand, overrideIfRegistered)) {
+            return this;
+        }
+
+        _registerCommand(baseCommand, true);
+
+        return this;
+    },
+
+    /**
+     * Registers all of the base commands available from the `and-cli` with the program.
+     * @param {boolean} [overrideIfRegistered=false] If true, subsequent registrations of a command
+     * with the same name will override the last. Otherwise, a warning will be displayed and the
+     * original command will remain.
+     *
+     * @returns `this` for chaining
+     */
+    registerBaseCommands(overrideIfRegistered = false) {
+        BASE_COMMAND_DEFINITIONS.forEach((commandDefinition) => {
+            const { command } = commandDefinition;
+
+            this.registerBaseCommand(command, overrideIfRegistered);
+        });
+
+        return this;
+    },
+
+    /**
+     * Register a single command with the program.
+     *
+     * @param {CommandDefinition} commandDefinition
+     * @param {boolean} [overrideIfRegistered=false] If true, subsequent registrations of a command
+     * with the same name will override the last. Otherwise, a warning will be displayed and the
+     * original command will remain.
+     *
+     * @returns `this` for chaining
+     */
+    registerCommand(commandDefinition, overrideIfRegistered = false) {
+        if (
+            !_validateCommandRegistration(
```

Comment:
Related to above, I suggest that this would simply call validate to verify whether or not the command can be registered, if false is returned, it would call `_removeCommand` and then `_registerCommand`.